sandbox/cgroup/freezer.go

```diff
@@ -178,6 +178,36 @@ func writeExistingFile(where string, data []byte, mode os.FileMode) error {
 	return errC
 }
 
+var freezeOneV2 = func(dir string) error {
+	groupName := filepath.Base(dir)
+	fname := filepath.Join(dir, "cgroup.freeze")
+	if err := writeExistingFile(fname, []byte("1")); err != nil {
+		if os.IsNotExist(err) {
+			//  the group may be gone already
+			return nil
+		}
+		return fmt.Errorf("cannot freeze processes in group %q: %v", groupName, err)
+	}
+	for i := 0; i < 30; i++ {
```

Comment:
Given that we are changing this, I think we should not do this anymore.

Instead we should:

- accept a context
- allow the caller to cancel the context
- poll indefinitely until the change is done or until the context is cancelled

Reply:
This way the caller can control the timeout and it's not so fragile anymore.

Reply:
I would like to add an internal timeout as well, so that if the caller doesn't pass a timeout context we still exit if cgroup is stuck in FREEZING state, having an upper bound of 5-10s.